src/query/api/v1/handler/influxdb/rewrite.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Author: Markus Stenberg <mstenber@aiven.io>
+ *
+ * Copyright (c) 2019 Aiven Oy
+ */
+
+package influxdb
+
+import "regexp"
+
+type regexpRewriter struct {
+	okRe          *regexp.Regexp
+	brokenStartRe *regexp.Regexp
+	brokenRestRe  *regexp.Regexp
+}
+
+func newRegexpRewriter(ok, brokenStart, brokenRest string) *regexpRewriter {
+	return &regexpRewriter{
+		okRe:          regexp.MustCompile(ok),
+		brokenStartRe: regexp.MustCompile(brokenStart),
+		brokenRestRe:  regexp.MustCompile(brokenRest),
+	}
+}
+
+func (self *regexpRewriter) ToValid(input []byte) []byte {
+	matched := self.okRe.Match(input)
+	// Fast path - costs only one regexp evaluation
+	// (This should be the common one)
+	if matched || len(input) == 0 {
+		return input
+	}
+	// Slow path - costs allocations etc
+	input = self.brokenStartRe.ReplaceAll(input, []byte("_"))
+	input = self.brokenRestRe.ReplaceAll(input, []byte("_"))
+	return input
+}
+
+// Stateful utility, which handles both __name__ ('metric') tag, as
+// well as rest of tags ('labels')
+//
+// It allow using any influxdb client, rewriting the tag names + the
+// magic __name__ tag to match what Prometheus expects
+type promRewriter struct {
+	Metric, Label *regexpRewriter
+}
+
+func newPromRewriter() *promRewriter {
+	return &promRewriter{
+		Metric: newRegexpRewriter("^[a-zA-Z_:][a-zA-Z0-9_:]*$",
```

Comment:
nit: Think this would be more straightforward to use if `promRewriter` didn't expose these, but instead had `validateMetric(..)` and `validateLabel(..)` methods